client/app/visualizations/edit-visualization-dialog.js

```diff
@@ -18,6 +18,20 @@ const EditVisualizationDialog = {
     this.onNewSuccess = this.resolve.onNewSuccess;
     this.visualization = copy(this.originalVisualization);
     this.visTypes = Visualization.visualizationTypes;
+    if (
+      this.visualization &&
+      isArray(this.query.visualizations) &&
+      (this.query.visualizations.length > 0)
+    ) {
+      // Forbid type change for first (min id) viz if it has type `TABLE`
+      this.canChangeType = !(
+        (this.visualization.type === 'TABLE') &&
+        (this.visualization.id === min(pluck(this.query.visualizations, 'id')))
+      );
```

Comment:
Maybe a simpler solution will be to not let the user change type after creating the visualization? I don't see much reason for a user to change type after creation and it will make it less of a guess work here :)

Reply:
Hm, makes sense. Will do :+1:

Reply:
Hi @deecay ! This code is outdated (@arikfr also didn't like it :smile: ) Now everything is much simpler :ok_hand: